geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseIndexAdapter.scala

```diff
@@ -579,6 +607,22 @@ object HBaseIndexAdapter extends LazyLogging {
       ds.connection.getBufferedMutator(params)
     }
 
+    import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType._
+    private val dtgIndex: Int = sft.flatMap(_.getDtgIndex).getOrElse(-1)
+
+    private val duration : Duration = sft.map{ sft =>
+      val userData = sft.getUserData
+      if (userData.containsKey("geomesa.feature.expiry")) {
+        Duration(userData.get("geomesa.feature.expiry").toString)
+      } else null
+    }.getOrElse(null)
+
+    private val expiration = if (duration != null && dtgIndex != -1) {
+      FeatureTimeExpiration("", dtgIndex, duration)
+    } else null
```

Comment:
you could replace this whole block with the existing convenience method:
```
import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
val expiration = indices.headOption.flatMap(_.sft.getFeatureExpiration).orNull
```

Reply:
that will support ingest time based expiration too, if someone wanted to use it (instead of only supporting based on an attribute time)

Reply:
this will change the configuration slightly - to use an attribute based expiration you'll need to configure it like `dtg(4 minutes)`. but that will be consistent with the docs and with accumulo

Reply:
I don't think this works because the goal is to have the TTL be based on a feature's dtg attribute; as far as I can tell this method's returned FeatureExpiration always returns a fixed TTL without considering feature dtg. I could be mistaken, but at the very least a simple one-to-one replacement of the code block breaks the tests.

Reply:
it should work if you configure the expiry in the unit test as `attribute(expiry)` i.e. `dtg(4 minutes)`

Reply:
Yeah, thanks - I only saw your first comment when I wrote my response; after editing the tests it passes now.